core/bones/numeric.py

```diff
@@ -80,7 +80,9 @@ def isEmpty(self, rawValue: Any):
             return True
         return rawValue == self.getEmptyValue()
 
-    def singleValueFromClient(self, value, skel, name, origData):
+    def singleValueFromClient(self, value: Any, skel: 'SkeletonInstance',
+                              bone_name: str, client_data: dict
+                              ) -> tuple[Any, list[ReadFromClientError] | None]:
```

Comment:
Is it necessary to provide full type annotations in sub-classes as well?
I would like to keep maintenance overhead low, if possible.